bika/lims/content/analysisspec.py

```diff
@@ -170,14 +172,23 @@ def getResultsRangeDict(self):
 
     security.declarePublic('getRemainingSampleTypes')
 
-    def getSampleTypes(self):
+    def getSampleTypes(self, active_only=True):
         """Return all sampletypes
         """
-        sampletypes = []
-        bsc = getToolByName(self, 'bika_setup_catalog')
-        for st in bsc(portal_type='SampleType', sort_on='sortable_title'):
-            sampletypes.append((st.UID, st.Title))
-
+        catalog = api.get_tool("bika_setup_catalog")
+        query = {
+            "portal_type": "SampleType",
+            # N.B. The `sortable_title` index sorts case sensitive. Since there
+            #      is no sort key for sample types, it makes more sense to sort
+            #      them alphabetically in the selection
+            "sort_on": "title",
+            "sort_order": "ascending"
+        }
+        results = catalog(query)
+        if active_only:
+            results = filter(api.is_active, results)
```

Comment:
Since `bika_setup_catalog` has the `inactive_state` index, you could simplify as follows:

```python
query = dict(portal_type="SampleType", sort_on="title", sort_order="ascending")
if active_only:
    query["inactive_state"] = "active"
results = api.search(query, "bika_setup_catalog")
```